tests/cli_integration.rs

```diff
@@ -1,506 +1,602 @@
-// cargo test --package surreal --bin surreal --no-default-features --features storage-mem --test cli_integration -- --nocapture
-
+// RUST_LOG=warn cargo make ci-cli-integration
```

Comment:
Skip to here during review: the rest of changes happen because I moved the tests into `mod cli_integration` so it's easier to scope down the `cargo test` command. I also changed how the server is started to account for the new `args` argument for the start_server helper